File: client/lib/services/auth_service.dart
import '../core/network/api_endpoints.dart';
import 'api_service.dart';
import 'storage_service.dart';

class AuthService {
  // ==========================
  // SIGNUP FLOW
  // ==========================

  // Step 1: Send OTP for signup
  static Future<Map<String, dynamic>> sendSignupOtp({
    required String name,
    required String email,
    required String phone,
    required String password,
  }) async {
    final response = await ApiService.post(
      url: ApiEndpoints.sendSignupOtp,
      body: {
        'name': name,
        'email': email,
        'phone': phone,
        'password': password,
      },
    );

    // Save signup data locally until OTP is verified
    if (response['success'] == true) {
      await StorageService.savePendingSignup({
        'name': name,
        'email': email,
        'phone': phone,
        'password': password,
      });
    }

    return response;
  }

  // Step 2: Verify OTP and create account
  static Future<Map<String, dynamic>> verifySignupOtp({
    required String otp,
  }) async {
    final pending = await StorageService.getPendingSignup();

    if (pending == null) {
      return {
        'success': false,
        'message': 'Signup session expired. Please try again.',
      };
    }

    final response = await ApiService.post(
      url: ApiEndpoints.verifySignupOtp,
      body: {
        'name': pending['name'],
        'email': pending['email'],
        'phone': pending['phone'],
        'password': pending['password'],
        'otp': otp,
      },
    );

    if (response['success'] == true) {
      if (response['token'] != null) {
        await StorageService.saveToken(response['token']);
      }
      if (response['user'] != null) {
        await StorageService.saveUser(response['user']);
      }
      // Clear pending data
      await StorageService.clearPendingSignup();
    }

    return response;
  }

  // Step 3: Resend signup OTP
  static Future<Map<String, dynamic>> resendSignupOtp() async {
    final pending = await StorageService.getPendingSignup();

    if (pending == null) {
      return {
        'success': false,
        'message': 'Signup session expired. Please try again.',
      };
    }

    return await ApiService.post(
      url: ApiEndpoints.resendSignupOtp,
      body: {
        'name': pending['name'],
        'email': pending['email'],
      },
    );
  }

  // ==========================
  // LOGIN
  // ==========================
  static Future<Map<String, dynamic>> login({
    required String email,
    required String password,
  }) async {
    final response = await ApiService.post(
      url: ApiEndpoints.login,
      body: {
        'email': email,
        'password': password,
      },
    );

    if (response['success'] == true) {
      if (response['token'] != null) {
        await StorageService.saveToken(response['token']);
      }
      if (response['user'] != null) {
        await StorageService.saveUser(response['user']);
      }
    }

    return response;
  }

  // ==========================
  // FORGOT PASSWORD FLOW
  // ==========================
  static Future<Map<String, dynamic>> forgotPassword({
    required String email,
  }) async {
    final response = await ApiService.post(
      url: ApiEndpoints.forgotPassword,
      body: {'email': email},
    );

    if (response['success'] == true) {
      await StorageService.saveResetEmail(email);
    }

    return response;
  }

  static Future<Map<String, dynamic>> verifyOtp({
    required String email,
    required String otp,
  }) async {
    return await ApiService.post(
      url: ApiEndpoints.verifyOtp,
      body: {'email': email, 'otp': otp},
    );
  }

  static Future<Map<String, dynamic>> resendOtp({
    required String email,
  }) async {
    return await ApiService.post(
      url: ApiEndpoints.resendOtp,
      body: {'email': email},
    );
  }

  static Future<Map<String, dynamic>> resetPassword({
    required String email,
    required String newPassword,
  }) async {
    final response = await ApiService.post(
      url: ApiEndpoints.resetPassword,
      body: {'email': email, 'newPassword': newPassword},
    );

    if (response['success'] == true) {
      await StorageService.clearResetEmail();
    }

    return response;
  }

  // LOGOUT
  static Future<void> logout() async {
    await StorageService.logout();
  }
}

File: admin-panel/src/services/authService.js
import axios from 'axios'
import { API_ENDPOINTS } from '../config/api'

// ==========================
// TOKEN MANAGEMENT
// ==========================
export const saveToken = (token) => localStorage.setItem('admin_token', token)
export const getToken = () => localStorage.getItem('admin_token')
export const removeToken = () => {
  localStorage.removeItem('admin_token')
  localStorage.removeItem('admin_data')
}

export const saveAdmin = (admin) => localStorage.setItem('admin_data', JSON.stringify(admin))
export const getAdmin = () => {
  const data = localStorage.getItem('admin_data')
  return data ? JSON.parse(data) : null
}

export const isAuthenticated = () => !!getToken()

// ==========================
// AXIOS INSTANCE
// ==========================
const api = axios.create()

api.interceptors.request.use((config) => {
  const token = getToken()
  if (token) {
    config.headers.Authorization = `Bearer ${token}`
  }
  return config
})

// ==========================
// SIGNUP FLOW
// ==========================
export const sendSignupOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.SEND_SIGNUP_OTP, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to send OTP',
    }
  }
}

export const verifySignupOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.VERIFY_SIGNUP_OTP, data)
    if (response.data.success) {
      saveToken(response.data.token)
      saveAdmin(response.data.admin)
    }
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Verification failed',
    }
  }
}

export const resendSignupOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.RESEND_SIGNUP_OTP, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to resend OTP',
    }
  }
}

// ==========================
// LOGIN
// ==========================
export const login = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.LOGIN, data)
    if (response.data.success) {
      saveToken(response.data.token)
      saveAdmin(response.data.admin)
    }
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Login failed',
    }
  }
}

// ==========================
// GOOGLE SIGN-IN
// ==========================
export const googleLoginCheck = async (email) => {
  try {
    const response = await axios.post(API_ENDPOINTS.GOOGLE_LOGIN_CHECK, { email })
    if (response.data.success) {
      saveToken(response.data.token)
      saveAdmin(response.data.admin)
    }
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Google login failed',
    }
  }
}

export const sendDeviceAuthorization = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.SEND_DEVICE_AUTH, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to send authorization',
    }
  }
}

export const checkDeviceStatus = async (email) => {
  try {
    const response = await axios.post(API_ENDPOINTS.CHECK_DEVICE_STATUS, { email })
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to check status',
    }
  }
}

// ✅ VERIFY GOOGLE SIGNUP OTP
export const verifyGoogleSignupOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.VERIFY_GOOGLE_SIGNUP_OTP, data)
    if (response.data.success) {
      saveToken(response.data.token)
      saveAdmin(response.data.admin)
    }
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Verification failed',
    }
  }
}

// ✅ RESEND GOOGLE SIGNUP OTP
export const resendGoogleSignupOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.RESEND_GOOGLE_SIGNUP_OTP, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to resend OTP',
    }
  }
}

// ==========================
// FORGOT PASSWORD FLOW
// ==========================
export const forgotPassword = async (email) => {
  try {
    const response = await axios.post(API_ENDPOINTS.FORGOT_PASSWORD, { email })
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to send OTP',
    }
  }
}

export const verifyOtp = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.VERIFY_OTP, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Invalid OTP',
    }
  }
}

export const resendOtp = async (email) => {
  try {
    const response = await axios.post(API_ENDPOINTS.RESEND_OTP, { email })
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Failed to resend',
    }
  }
}

export const resetPassword = async (data) => {
  try {
    const response = await axios.post(API_ENDPOINTS.RESET_PASSWORD, data)
    return response.data
  } catch (error) {
    return {
      success: false,
      message: error.response?.data?.message || 'Reset failed',
    }
  }
}

// ==========================
// LOGOUT
// ==========================
export const logout = () => {
  removeToken()
  window.location.href = '/login'
}